lib/block/header.ex

```diff
@@ -284,13 +306,35 @@ defmodule Block.Header do
       ...>  %Block.Header{number: 3_000_000, timestamp: 55, difficulty: 300_000}
       ...> )
       268_734_142
+
+      Test actual Ropsten genesis block
+      iex> Block.Header.get_difficulty(
+      ...>   %Block.Header{number: 0, timestamp: 0},
+      ...>   nil,
+      ...>   0x100000,
+      ...>   0x020000,
+      ...>   0x0800,
+      ...>   0
+      ...> )
+      1_048_576
+
+      # Test actual Ropsten first block
+      iex> Block.Header.get_difficulty(
+      ...>   %Block.Header{number: 1, timestamp: 1_479_642_530},
+      ...>   %Block.Header{number: 0, timestamp: 0, difficulty: 1_048_576},
+      ...>   0x100000,
+      ...>   0x020000,
+      ...>   0x0800,
+      ...>   0
+      ...> )
+      997_888
   """
-  @spec get_difficulty(t, t | nil) :: integer()
-  def get_difficulty(header, parent_header) do
+  @spec get_difficulty(t, t | nil, integer()) :: integer()
+  def get_difficulty(header, parent_header, initial_difficulty \\ @initial_difficulty, minimum_difficulty \\ @minimum_difficulty, difficulty_bound_divisor \\ @difficulty_bound_divisor, homestead_block \\ @homestead_block) do
     cond do
-      header.number == 0 -> @initial_difficulty
-      is_before_homestead?(header) -> max(@initial_difficulty, parent_header.difficulty + difficulty_x(parent_header.difficulty) * difficulty_s1(header, parent_header) + difficulty_e(header))
-      true -> max(@initial_difficulty, parent_header.difficulty + difficulty_x(parent_header.difficulty) * difficulty_s2(header, parent_header) + difficulty_e(header))
+      header.number == 0 -> initial_difficulty
+      is_before_homestead?(header, homestead_block) -> max(minimum_difficulty, parent_header.difficulty + difficulty_x(parent_header.difficulty, difficulty_bound_divisor) * difficulty_s1(header, parent_header) + difficulty_e(header))
```

Comment:
any way we could break these two lines up a bit while we're in here?  Is there a more descriptive name we could give  `difficulty_x`  and friends or are those just variables in the difficulty formula?

Reply:
Those are the names from the Yellow Paper:

<img width="389" alt="screen shot 2017-09-09 at 7 19 47 pm" src="https://user-images.githubusercontent.com/271024/30245595-e0746ad8-9593-11e7-92ce-4eeed401e14d.png">

It's not entirely clear to me the right way to describe them, since I just copied the formulas directly as is.

Reply:
Yep, looks like they're just equations :/ ah well ¯\_(ツ)_/¯